src/Confluent.Kafka/Producer.cs

```diff
@@ -52,9 +52,10 @@ public class Producer : IDisposable
         private Task StartPollTask(CancellationToken ct)
             => Task.Factory.StartNew(() =>
                 {
-                    while (!ct.IsCancellationRequested)
+                    while (true)
                     {
                         this.kafkaHandle.Poll((IntPtr)POLL_TIMEOUT_MS);
+                        ct.ThrowIfCancellationRequested();
```

Comment:
move this line above the poll since the task may have already been cancelled by the time it comes to the first iteration (from MS example)

Reply:
good point